sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapi/AADResourceServerOboConfigurationTest.java

```diff
@@ -24,6 +26,30 @@
             "azure.activedirectory.client-id=fake-client-id",
             "azure.activedirectory.client-secret=fake-client-secret");
 
+    @Test
+    public void testWithoutAnyPropertiesSet() {
+        new WebApplicationContextRunner()
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .run(context -> {
+                assertThat(context).doesNotHaveBean(AADAuthenticationProperties.class);
+                assertThat(context).doesNotHaveBean(ClientRegistrationRepository.class);
+                assertThat(context).doesNotHaveBean(OAuth2AuthorizedClientRepository.class);
+            });
+    }
```

Comment:
We'd better add more tests: 
1. `client-id` configured, but no `authorization-clients`.
2. `client-id` not configured, but `authorization-clients` configured.

Reply:
Yes, I'm just about to add them.

Reply:
Current tests can not make sure `@ConditionalOnExpression("!'${azure.activedirectory.authorization-clients}'.isEmpty()")` take effect.